packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -254,12 +259,34 @@ function makeAugmentedSchema(
     const relationshipFields = new Map<string, ObjectFields>();
 
     relationshipProperties.forEach((relationship) => {
+        constants.RESERVED_TYPE_NAMES.forEach(([label, message]) => {
+            let toThrowError = false;
+
+            if (label === "Connection" && relationship.name.value.endsWith("Connection")) {
+                toThrowError = true;
+            }
+
+            if (relationship.name.value === label) {
+                toThrowError = true;
+            }
+
+            if (toThrowError) {
+                throw new Error(message);
+            }
+        });
```

Comment:
This could probably be put into a function (exact duplicate of above), but not necessary now. 🙂